crypto/evp/keymgmt_lib.c

```diff
@@ -370,3 +370,82 @@ void *evp_keymgmt_util_gen(EVP_PKEY *target, EVP_KEYMGMT *keymgmt,
 
     return keydata;
 }
+
+struct export_todata_st {
+    OSSL_PARAM *params;
+};
+
+/*
+ * Some keys have multiple instances, so mark them as used so that they are only
+ * found once.
+ */
+static OSSL_PARAM *locate_used(const OSSL_PARAM *params, const char *key,\
+                               int used[], int used_len)
+{
+    int i;
+    OSSL_PARAM *p = (OSSL_PARAM *)params;
+
+    for (i = 0; p->key != NULL && i < used_len; p++, i++) {
+        if (!used[i] && strcmp(key, p->key) == 0) {
```

Comment:
Ugh :(

Reply:
RSA is a good example of something with multiple fields that are the same, which is particularly ugly (i.e- how do we know how many it has before we do the call).

Reply:
> (i.e- how do we know how many it has before we do the call).

find out by getting the value of the "primes" (**SSL_PKEY_PARAM_RSA_PRIMES**)

Reply:
No!

When we get a new algorithm that requires _n_ happy numbers to work, asking for primes isn't useful.

Reply:
But supposedly, *the application* knows what to ask for.

"Know your provider" -- me